goopicsi.go

```diff
@@ -225,6 +226,68 @@ func NVMeControllerDisconnect(id int64) error {
 	return nil
 }
 
+// ExposeRemoteNVMe creates a new NVMe Subsystem and NVMe controller
+func ExposeRemoteNVMe(subsystemID string, subsystemNQN string, maxNamespaces int64, controllerID string) error {
+	if conn == nil {
+		err := dialConnection()
+		if err != nil {
+			return err
+		}
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	client := pb.NewFrontendNvmeServiceClient(conn)
+	data1, err := client.GetNVMeSubsystem(ctx, &pb.GetNVMeSubsystemRequest{SubsystemId: &pbc.ObjectKey{Value: subsystemID}})
+	if err != nil {
+		log.Printf("No existing NVMe Subsystem found with subsystemID: %v", subsystemID)
+	}
+
+	if data1 == nil {
+		response1, err := client.CreateNVMeSubsystem(ctx, &pb.CreateNVMeSubsystemRequest{
+			Subsystem: &pb.NVMeSubsystem{
+				Spec: &pb.NVMeSubsystemSpec{
+					Id:            &pbc.ObjectKey{Value: subsystemID},
```

Comment:
you should generate subsystemID here using, for example `uuid.New()` user of the library should not care about IDs... he is only aware of NQN...